parser_library/src/workspaces/parse_lib_provider.h

```diff
@@ -37,6 +37,8 @@ class parse_lib_provider
 
     virtual bool has_library(const std::string& library, context::hlasm_context& hlasm_ctx) const = 0;
 
+    virtual std::map<std::string, std::string> get_asm_options(const std::string&) = 0;
```

Comment:
The more I think about this, the more unsure I am as to why to add this new member function into this particular interface.

Reply:
It was added, to pass the SYSPARM value in the hlasm_context() constructor, dint get a better way to directly access workpace .cpp

Reply:
There is quite a few parameters on the analyzer already. I am wondering whether a new struct `analyzer_options` should be added. And that would contain collect_hl_info, tracer, library_data and a new members related to sysparm and profile parameters.
What do you think @michalbali256 ?

Reply:
I thought about this too and I think it is a good place to add the function. Moving forward, the interface should become more like "processor group provider" instead of just "library provider".

Currently, I do not even see a reason why should the `workspace` implement the interface instead of `workspaces::processor_group`, since every method first finds the correct processor group (that could be found just once when the parsing starts) and then uses members of the processor group to provide the library. But I see that as a refactor for another time.

Reply:
I couldn't see your latest answer as I was typing mine. Yes, it is a good idea for the future, since it is easier to manage a struct than several constructors with every change. However I still think adding the new virtual function into the `parse_lib_provider` is a good change

Reply:
Will the function still make sense if we are able to pass the options directly to the analyzer?

Reply:
I think so. A new parameter would require a change in processor_file_impl::parse signature to take one more parameter excapt `parse_lib_provider`. Also when we will implement additional options like profile, it will be possible to get them from deeper layers of analyzer which already have reference to `parse_lib_provider`

If there will be additional options not tied with processor groups that we need to pass into analyzer, we should revisit this conversation in the future.

Reply:
So far I only see two callers of `parse`, both in workspace...